src/routes/stripeRoute.ts

```diff
@@ -0,0 +1,15 @@
+import { Router } from 'express';
+import stripeController from '../modules/stripe/controller/stripeController';
+import { gatewayAuthentication } from '../middlewares/authenticationMiddleware';
+import { authorizationSchema } from '../modules/auth/validation/authValidation';
+import { isBodyValidation, isHeaderValidation, isQueryValidation } from '../middlewares/requestMiddleware';
+import { planDetailsSchema, checkoutSessionSchema, emailSchema } from '../modules/stripe/validation/stripeValidation';
+
+const router: Router = Router();
+
+router.get('/products', isHeaderValidation(authorizationSchema), gatewayAuthentication, stripeController.stripeGetProducts);
+router.post('/product', isHeaderValidation(authorizationSchema), gatewayAuthentication, isBodyValidation(planDetailsSchema), stripeController.stripecreateProduct);
+router.get('/subscriptions', isHeaderValidation(authorizationSchema), gatewayAuthentication, isQueryValidation(emailSchema), stripeController.stripeGetSubscription);
+router.post('/checkout-session', isHeaderValidation(authorizationSchema), gatewayAuthentication, isBodyValidation(checkoutSessionSchema), stripeController.stripeCheckoutSession);
```

Comment:
isHeaderValidation and gatewayAuthentication are used in every route, so in this case, you should add this to the main route ('/stripe'). Same should be true for other routes that use same validations

Reply:
Okay, noted